src/ripple/app/transactors/SetSignerList.cpp

```diff
@@ -0,0 +1,322 @@
+            deprecatedLogs().journal("SetSignerList"))
+    {
+
+    }
+
+    /**
+    Applies the transaction if it is well formed and the ledger state permits.
+    */
+    TER doApply () override;
+
+private:
+    TER replaceSignerList (std::uint32_t quorum, uint256 const& index);
+    TER destroySignerList (uint256 const& index);
+
+    // signers are not const because method (intentionally) sorts vector.
+    TER validateQuorumAndSignerEntries (
+        std::uint32_t quorum,
+        std::vector<SignerEntries::SignerEntry>& signers) const;
+
+    void writeSignersToLedger (
+        SLE::pointer ledgerEntry,
+        std::uint32_t quorum,
+        std::vector<SignerEntries::SignerEntry> const& signers);
+
+    static std::size_t ownerCountDelta (std::size_t entryCount);
+};
+
+//------------------------------------------------------------------------------
+
+TER SetSignerList::doApply ()
+{
+    assert (mTxnAccount);
+
+    // All operations require our ledger index.  Compute that once and pass it
+    // to our handlers.
+    uint256 const index = getSignerListIndex (mTxnAccountID);
+
+    // Check the quorum.  A non-zero quorum means we're creating or replacing
+    // the list.  A zero quorum means we're destroying the list.
+    std::uint32_t const quorum (mTxn.getFieldU32 (sfSignerQuorum));
+
+    bool const hasSignerEntries (mTxn.isFieldPresent (sfSignerEntries));
+    if (quorum && hasSignerEntries)
+        return replaceSignerList (quorum, index);
+
+    if ((quorum == 0) && !hasSignerEntries)
+        return destroySignerList (index);
+
+    if (m_journal.trace) m_journal.trace <<
+        "Malformed transaction: Invalid signer set list format.";
+    return temMALFORMED;
+}
+
+TER
+SetSignerList::replaceSignerList (std::uint32_t quorum, uint256 const& index)
+{
+    if (!mTxn.isFieldPresent (sfSignerEntries))
+    {
+        if (m_journal.trace) m_journal.trace <<
+            "Malformed transaction: Need signer entry array.";
+        return temMALFORMED;
+    }
+
+    SignerEntries::Decoded signers (
+        SignerEntries::deserialize (mTxn, m_journal, "transaction"));
+
+    if (signers.ter != tesSUCCESS)
+        return signers.ter;
+
+    // Validate our settings.
+    if (TER const ter = validateQuorumAndSignerEntries (quorum, signers.vec))
+        return ter;
+
+    // This may be either a create or a replace.  Preemptively destroy any
+    // old signer list.  May reduce the reserve, so this is done before
+    // checking the reserve.
+    if (TER const ter = destroySignerList (index))
+        return ter;
+
+    // Compute new reserve.  Verify the account has funds to meet the reserve.
+    std::size_t const oldOwnerCount = mTxnAccount->getFieldU32 (sfOwnerCount);
+    std::size_t const addedOwnerCount = ownerCountDelta (signers.vec.size ());
+
+    std::uint64_t const newReserve =
+        mEngine->getLedger ()->getReserve (oldOwnerCount + addedOwnerCount);
+
+    if (mPriorBalance.getNValue () < newReserve)
```

Comment:
Great question, and I'd forgotten this detail.  I modeled this implementation off of `TicketCreate`, which allows the fee to come out of the reserve.  But I should certainly add a comment to that effect.